### PR TITLE
Handle case-sensitive attributes like viewBox.

### LINKED ISSUE #374: `viewBox` not recognized in lit-extended
When using the `$=` syntax to dynamically update the `viewBox` attribute of an SVG, the following error appears in the console and the DOM does not render:

> Uncaught TypeError: Cannot read property 'value' of null

[This Plunkr](https://next.plnkr.co/edit/CEjDaef5AR2I1Wra) concisely demonstrates the issue.

For reference, here is the SVG snippet:

```
<svg viewBox$="0 0 ${viewBoxSize} ${viewBoxSize}" width$="100px" height$="100px">
    ${ svg`<rect x="0" y="0" width$="${viewBoxSize}" height$="${viewBoxSize}"/>` }
</svg>
```

and the import for lit-extended:

```
import {html, svg, render} from 'https://unpkg.com/lit-html@0.10.2/lib/lit-extended.js';
```

This is related to #275 but as can be seen by the Plunkr it is a different issue.

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #374

### CODE REVIEW

**Comment by intcreator** [comment]
Looking good so far!

File: src/test/lit-html_test.ts [hunk 1]
```diff
@@ -324,6 +315,14 @@ suite('lit-html', () => {
         assert.equal(stripExpressionDelimeters(container.innerHTML), '<div foo="bar"></div>');
       });
 
+      test('renders a case-sensitive attribute', () => {
+        const size = 100;
+        render(html`<svg viewBox="0 0 ${size} ${size}"></svg>`, container);
+        assert.equal(
+            stripExpressionDelimeters(container.innerHTML),
+            '<svg viewBox="0 0 100 100"></svg>');
+      });
```
Comment (kevinpschaaf) [inline src/test/lit-html_test.ts]: For Edge test to pass, apparently need to strip the extra `xmlns` attribute when asserting.  And they helpfully render the innerHTML as a self-closing tag, apparently.  Sigh.
```
expected '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100" />' to equal '<svg viewBox="0 0 100 100"></svg>'
```
Maybe just change it to 
```js
assert.match(stripExpressionDelimeters(container.innerHTML), /viewBox="0 0 100 100"/);
```

File: src/lit-html.ts [hunk 2]
```diff
@@ -320,16 +327,21 @@ export class Template {
...
               lastAttributeNameRegex.exec(stringForPart)![1];
           // Find the corresponding attribute
-          // TODO(justinfagnani): remove non-null assertion
-          const attribute = attributes.getNamedItem(attributeNameInPart)!;
-          const stringsForAttributeValue = attribute.value.split(markerRegex);
+          // If the attribute name contains special characters, lower-case it
+          // so that on XML nodes with case-sensitive getAttribute() we can
+          // still find the attribute, which will have been lower-cased by
+          // the parser.
+          const attributeLookupName = /^[a-zA-z-]*$/.test(attributeNameInPart) ?
```
Comment (jridgewell) [inline src/lit-html.ts]: The `A-z` looks like a bug.

Comment (justinfagnani) [reply src/lit-html.ts]: Definitely. Nice catch

**Pull request merged**